### PR TITLE
[pkg/opentelemetrytransformationlanguage] Rename TQL to OTTL

### LINKED ISSUE #13919: Rename "TQL" (telemetry query language) for better clarity while it's still painless to do so
Not news: there's a sensible and valuable domain-specific language (DSL) built into the OTel collector that basically takes telemetry data and transforms it into other telemetry data. All good and makes sense. 👍

The "bug" here is that the DSL in question is called the "Telemetry Query Language" (aka "TQL" – see more [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/telemetryquerylanguage/tql))... when I heard about it, I assumed it was a language that performs *analysis* of the telemetry data... presumably due to the naming association with SQL? I brought this up with the @open-telemetry/governance-committee last week and others had the same concern.

Along those lines, can we come up with a different name before this one is too hard to change? I am glad to make suggestions (e.g., "Telemetry Transformation Language" or "Telemetry Transformation Expression Language"), but I'm not picky as long as it's clear that this is not an *analytical* query language.

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Renames TQL to OTTL.  No functional changes

**Link to tracking Issue:** 
Resolves #13919

**Testing:** 
Unit tests

**Documentation:**
All documentation/links have been updated

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
Not opposed to shortening the package directory to `oteltransformationlanguage`

**Comment by jpkrohling** [comment]
cc @kovrus

File: pkg/opentelemetrytransformationlanguage/functions/ottlcommon/README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ The `set` function allows users to set a telemetry field using a value.
 
 `target` is a path expression to a telemetry field. `value` is any value type. If `value` resolves to `nil`, e.g. it references an unset map value, there will be no action.
 
-How the underlying telemetry field is updated is decided by the path expression implementation provided by the user to the `tql.ParseQueries`.
+How the underlying telemetry field is updated is decided by the path expression implementation provided by the user to the `OTTL.ParseQueries`.
```
Comment (evan-bradley) [inline pkg/opentelemetrytransformationlanguage/functions/ottlcommon/README.md:83]:
```suggestion
How the underlying telemetry field is updated is decided by the path expression implementation provided by the user to the `ottl.ParseQueries`.
```

File: pkg/opentelemetrytransformationlanguage/functions/ottlotel/func_delete_matching_keys.go [hunk 2]
```diff
@@ -12,23 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package tqlotel // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/functions/tqlotel"
+package ottlotel // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/opentelemetrytransformationlanguage/functions/OTTLotel"
```
Comment (evan-bradley) [inline pkg/opentelemetrytransformationlanguage/functions/ottlotel/func_delete_matching_keys.go:15]:
```suggestion
package ottlotel // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/opentelemetrytransformationlanguage/functions/ottlotel"
```

**Review by evan-bradley — COMMENTED** [review]
I found a handful of places where lowercase `tql` is replaced with uppercase `OTTL`. There are a handful of places where `OTTLotel` is specified in import paths where I didn't comment on every one. I've highlighted once instance, it should be easy to find and replace the rest.

As for renaming the directory to `pkg/oteltransformationlanguage`, I think that would make sense since the current name is so long. Abbreviations are common in component names (e.g. `k8s` or `attrs`), and `otel` is already used in `internal/otelcontribcore`, so I think there's some precedent for it as well.

**Comment by TylerHelmuth** [comment]
@evan-bradley I think I address all your feedback.

File: processor/transformprocessor/internal/traces/processor.go [hunk 3]
```diff
@@ -21,24 +21,24 @@ import (
...
 		functions,
-		tqltraces.ParsePath,
-		tqltraces.ParseEnum,
-		common.NewTQLLogger(settings.Logger),
+		ottltraces.ParsePath,
+		ottltraces.ParseEnum,
+		common.NewOTTLLogger(settings.Logger),
 	)
-	queries, err := tqlp.ParseQueries(statements)
+	queries, err := ottlp.ParseQueries(statements)
```
Comment (evan-bradley) [inline processor/transformprocessor/internal/traces/processor.go:41]: Likely outside the scope of this PR, but should the `Query` type be renamed now that it is no longer the TQL? I think `Statement` would work well, but there may be better alternatives.

File: processor/routingprocessor/README.md [hunk 4]
```diff
@@ -94,9 +94,9 @@ It is also possible to use both the conventional routing items configuration and
 - [OTTL] expressions can be applied only to resource attributes.
 - Currently, it is not possible to specify the boolean expression without function invocation as the routing condition. It is required to provide the NOOP `route()` or any other supported function as part of the routing expression, see [#13545](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13545) for more information.
 - Supported [OTTL] functions:
-  - [IsMatch](../../pkg/telemetryquerylanguage/functions/tqlcommon/README.md#IsMatch)
-  - [delete_key](../../pkg/telemetryquerylanguage/functions/tqlotel/README.md#delete_key)
-  - [delete_matching_keys](../../pkg/telemetryquerylanguage/functions/tqlotel/README.md#delete_matching_keys)
+  - [IsMatch](../../pkg/oteltransformationlanguage/functions/tqlcommon/README.md#IsMatch)
```
Comment (evan-bradley) [inline processor/routingprocessor/README.md:97]:
```suggestion
  - [IsMatch](../../pkg/oteltransformationlanguage/functions/ottlcommon/README.md#IsMatch)
```

File: processor/routingprocessor/README.md [hunk 5]
```diff
@@ -94,9 +94,9 @@ It is also possible to use both the conventional routing items configuration and
...
+  - [delete_key](../../pkg/oteltransformationlanguage/functions/tqlotel/README.md#delete_key)
```
Comment (evan-bradley) [inline processor/routingprocessor/README.md:98]:
```suggestion
  - [delete_key](../../pkg/oteltransformationlanguage/functions/ottlotel/README.md#delete_key)
```

**Review by evan-bradley — COMMENTED** [review]
I found a few additional replacements needed in a readme, but otherwise looks good.

File: processor/transformprocessor/internal/traces/processor.go (hunk 3 above)
Comment (TylerHelmuth) [reply processor/transformprocessor/internal/traces/processor.go:41]: Yes it should be changed. It'll do that in a follow up PR

**Comment by TylerHelmuth** [comment]
@evan-bradley @bogdandrutu @kentquirk @kovrus please take a look.

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**